Lantiq Image: use the name from CONFIG_VERSION_IMAGE if provided, otherwise the previous default

diff --git a/include/version.mk b/include/version.mk
--- a/include/version.mk
+++ b/include/version.mk
@@ -44,6 +44,8 @@ VERSION_PRODUCT:=$(if $(VERSION_PRODUCT)
 VERSION_HWREV:=$(call qstrip_escape,$(CONFIG_VERSION_HWREV))
 VERSION_HWREV:=$(if $(VERSION_HWREV),$(VERSION_HWREV),v0)
 
+VERSION_IMAGE:=$(call qstrip,$(CONFIG_VERSION_IMAGE))
+
 define taint2sym
 $(CONFIG_$(firstword $(subst :, ,$(subst +,,$(subst -,,$(1))))))
 endef
@@ -85,4 +87,21 @@ VERSION_SED:=$(SED) 's,%U,$(VERSION_REPO
 	-e 's,%P,$(VERSION_PRODUCT),g' \
 	-e 's,%h,$(VERSION_HWREV),g'
 
+VERSION_IMAGE_SED:=$(shell echo $(VERSION_IMAGE) | $(STAGING_DIR_HOST)/bin/sed \
+	-e 's,%V,$(VERSION_NUMBER),g' \
+	-e 's,%v,\L$(subst $(space),_,$(VERSION_NUMBER)),g' \
+	-e 's,%C,$(VERSION_CODE),g' \
+	-e 's,%c,\L$(subst $(space),_,$(VERSION_CODE)),g' \
+	-e 's,%N,$(VERSION_NICK),g' \
+	-e 's,%n,\L$(subst $(space),_,$(VERSION_NICK)),g' \
+	-e 's,%D,$(VERSION_DIST),g' \
+	-e 's,%d,\L$(subst $(space),_,$(VERSION_DIST)),g' \
+	-e 's,%R,$(REVISION),g' \
+	-e 's,%T,$(BOARD),g' \
+	-e 's,%S,$(BOARD)/$(if $(SUBTARGET),$(SUBTARGET),generic),g' \
+	-e 's,%t,$(VERSION_TAINTS),g' \
+	-e 's,%M,$(VERSION_MANUFACTURER),g' \
+	-e 's,%P,$(VERSION_PRODUCT),g' \
+	-e 's,%h,$(VERSION_HWREV),g')
+
 VERSION_SED_SCRIPT:=$(subst '\'','\'\\\\\'\'',$(VERSION_SED))
diff --git a/target/linux/lantiq/image/Makefile b/target/linux/lantiq/image/Makefile
--- a/target/linux/lantiq/image/Makefile
+++ b/target/linux/lantiq/image/Makefile
@@ -10,12 +10,17 @@
 
 include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/image.mk
+include $(INCLUDE_DIR)/version.mk
 
 JFFS2_BLOCKSIZE = 64k 128k 256k
 KDIR_TMP:=$(KDIR)/tmp
 
 LOADER_MAKE := $(NO_TRACE_MAKE) -C lzma-loader KDIR=$(KDIR)
 
+IMAGE_NAME_DEFAULT:='MIPS OpenWrt Linux-$(LINUX_VERSION)'
+
+IMAGE_NAME:=$(if $(VERSION_IMAGE_SED),$(VERSION_IMAGE_SED),$(IMAGE_NAME_DEFAULT))
+
 define Image/BuildLoader/Template
 	-rm -rf $(KDIR)/lzma-loader
 	$(LOADER_MAKE) LOADER=loader$(2)-$(1).bin\
@@ -45,7 +50,7 @@ endef
 
 define MkImageLzma
 	mkimage -A mips -O linux -T kernel -a 0x80002000 -C lzma \
-		-e 0x80002000 -n 'MIPS OpenWrt Linux-$(LINUX_VERSION)' \
+		-e 0x80002000 -n $(IMAGE_NAME) \
 		-d $(KDIR)/vmlinux$(2)-$(1).lzma $(KDIR)/uImage-$(1)$(2)
 endef
 
@@ -73,7 +78,7 @@ endef
 
 define MkImageGzip
 	mkimage -A mips -O linux -T kernel -a 0x80002000 -C gzip \
-		-e 0x80002000 -n 'MIPS OpenWrt Linux-$(LINUX_VERSION)' \
+		-e 0x80002000 -n $(IMAGE_NAME) \
 		-d $(KDIR)/vmlinux$(2)-$(1).gzip $(KDIR)/uImage-$(1)$(2)
 endef
 
